xwiki-rendering-macros/xwiki-rendering-macro-content/src/main/java/org/xwiki/rendering/internal/macro/content/ContentMacro.java

```diff
@@ -78,52 +84,45 @@ public class ContentMacro extends AbstractMacro<ContentMacroParameters>
      */
     public ContentMacro()
     {
-        super("Content", DESCRIPTION, new DefaultContentDescriptor(CONTENT_DESCRIPTION, true,
-            Block.LIST_BLOCK_TYPE), ContentMacroParameters.class);
+        super("Content", DESCRIPTION, new DefaultContentDescriptor(CONTENT_DESCRIPTION, false, Block.LIST_BLOCK_TYPE),
+            ContentMacroParameters.class);
+
         setDefaultCategories(Set.of(DEFAULT_CATEGORY_CONTENT));
     }
 
     @Override
     public boolean supportsInlineMode()
     {
-        return false;
+        return true;
     }
 
     @Override
-    public List<Block> execute(ContentMacroParameters parameters, String content, MacroTransformationContext context)
-        throws MacroExecutionException
+    public List<Block> execute(ContentMacroParameters parameters, String macroContent,
+        MacroTransformationContext context) throws MacroExecutionException
     {
-        try {
-            List<Block> blocks = getSyntaxParser(parameters.getSyntax()).parse(new StringReader(content)).getChildren();
-            MetaDataBlock metaDataBlock = new MetaDataBlock(blocks, MetaData.SYNTAX, parameters.getSyntax());
-
-            metaDataBlock.getMetaData().addMetaData(this.getNonGeneratedContentMetaData());
-            return Collections.singletonList(metaDataBlock);
-        } catch (ParseException e) {
-            throw new MacroExecutionException(
-                String.format("Failed to parse macro content in syntax [%s]", parameters.getSyntax()), e);
+        Syntax syntax = parameters.getSyntax();
+        String content = macroContent;
+        MetaData metadata = new MetaData();
+        if (syntax != null) {
+            // Remember the custom syntax as it can be needed for sub macros
+            metadata.addMetaData(MetaData.SYNTAX, syntax);
         }
-    }
-
-    /**
-     * Get the parser for the passed Syntax.
-     *
-     * @param syntax the Syntax for which to find the Parser
-     * @return the matching Parser that can be used to parse content in the passed Syntax
-     * @throws MacroExecutionException if there's no Parser in the system for the passed Syntax
-     */
-    protected Parser getSyntaxParser(Syntax syntax) throws MacroExecutionException
-    {
-        ComponentManager componentManager = this.componentManagerProvider.get();
-        if (componentManager.hasComponent(Parser.class, syntax.toIdString())) {
-            try {
-                return componentManager.getInstance(Parser.class, syntax.toIdString());
-            } catch (ComponentLookupException e) {
-                throw new MacroExecutionException(
-                    String.format("Failed to lookup Parser for syntax [%s]", syntax.toIdString()), e);
+        if (parameters.getSource() != null) {
```

Comment:
I'm wondering if this is interesting to send an error if a source parameter and a macro content are defined at the same time (to remove the ambiguity as iirc for now the source param takes priority).

Reply:
I'm not sure it brings that much value to have an error (it's not that mysterious that the source have priority, IMO), and it could be annoying while the content is a work in progress.